src/core/common/frame_builder.hpp

```diff
@@ -228,6 +228,40 @@ class FrameBuilder
         return AppendBytes(&aObject, sizeof(ObjectType));
     }
 
+    /**
+     * Appends the given number of bytes to the `FrameBuilder`.
+     *
+     * This method reserves @p aLength bytes at the current position of the `FrameBuilder` and returns a pointer to the
+     * start of this reserved buffer if successful. The reserved bytes are left uninitialized. The caller is
+     * responsible for initializing them.
+     *
+     * @param[in] aLength  The number of bytes to append.
+     *
+     * @returns A pointer to the start of the appended bytes if successful, or `nullptr` if there are not enough
+     *          remaining bytes to append @p aLength bytes.
+     *
+     */
+    void *AppendLength(uint16_t aLength);
+
+    /**
+     * Appends an object to the `FrameBuilder`.
+     *
+     * @tparam    ObjectType  The object type to append.
+     *
+     * This method reserves bytes in `FrameBuilder`  to accommodate an `ObjectType` and returns a pointer to the
+     * appended `ObjectType`. The `ObjectType` bytes are left uninitialized. Caller is responsible to initialize them.
+     *
+     * @returns A pointer the appended `ObjectType` if successful, or `nullptr` if there are not enough remaining
+     *          bytes to append an `ObjectType`.
+     *
+     */
+    template <typename ObjectType> ObjectType *Append(void)
```

Comment:
We use it now when appending seq number or command id.
```c++
    if (mType == kTypeMacCmd)
    {
        builder.Append<uint8_t>(); // Placeholder for Command ID
    }
```

It will be used with more complex types in follow-up PR simplifying header IEs.